### PR TITLE
Make the range classes use a static inner class

### PR DESCRIPTION
**Pull request by jponge**
This is generally better to use static vs instance-attached inner classes (performance, GC, etc).

Another point is that this makes the Eclipse compiler happy due to a bug it has.

### CODE REVIEW

**Comment by jponge** [comment]
Ping @yloiseau for review + merge

**Comment by yloiseau** [comment]
My bad...

File: src/main/java/gololang/CharRange.java [hunk 1]
```diff
@@ -37,10 +37,10 @@ Character defaultValue() {
 
   @Override
   public int size() {
```
Comment (yloiseau) [inline src/main/java/gololang/CharRange.java]: Couldn't we now generalize by moving `size` into `AbstractRange`

Comment (jponge) [inline src/main/java/gololang/CharRange.java]: Indeed!

Le dim. 11 oct. 2015 à 11:23, Yannick Loiseau <notifications@github.com> a
écrit :

**Pull request merged**